### ISSUE TITLE
UserWarning: An issue occurred while importing 'pyg-lib'. Disabling its usage. Stacktrace: /lib64/libm.so.6: version `GLIBC_2.27' not found

### ISSUE THREAD

**Issue by Dengsongquanch** [pyg-team/pyg-lib#363]
### 🐛 Describe the bug

when I import pyg_lib,there occurs an error:
UserWarning: An issue occurred while importing 'pyg-lib'. Disabling its usage. Stacktrace: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /public/home/yika/miniforge3/envs/deepH3/lib/python3.9/site-packages/libpyg.so)
I found that upgrade glibc can solve this problem. But glibc is a fundamental package, upgrade it may cause some very serious errors. So I want to know if there are any other solutions available.

### Environment

* `pyg-lib` version:0.4.0+pt112cu116
* PyTorch version:1.12.0+cu116
* OS:CentOS Linux 7 (Core)
* Python version:3.9
* CUDA/cuDNN version:cuda11.6，cudnn8.3.2
* How you installed PyTorch and `pyg-lib` (`conda`, `pip`, source):conda
* Any other relevant information:

**Comment by Dengsongquanch** [comment]
My environment is below
# packages in environment at /public/home/yika/miniforge3/envs/deepH3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
absl-py                   2.1.0                    pypi_0    pypi
aiohappyeyeballs          2.4.3                    pypi_0    pypi
aiohttp                   3.10.10                  pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
async-timeout             4.0.3                    pypi_0    pypi
attrs                     24.2.0                   pypi_0    pypi
aws-c-auth                0.7.4                h1083cbe_2    conda-forge
aws-c-cal                 0.6.2                h09139f6_2    conda-forge
aws-c-common              0.9.3                hd590300_0    conda-forge
aws-c-compression         0.2.17               h184a658_3    conda-forge
aws-c-event-stream        0.3.2                h6fea174_2    conda-forge
aws-c-http                0.7.13               hb59894b_2    conda-forge
aws-c-io                  0.13.33              h161b759_0    conda-forge
aws-c-mqtt                0.9.7                h55cd26b_0    conda-forge
aws-c-s3                  0.3.17               hfb4bb88_4    conda-forge
aws-c-sdkutils            0.1.12               h184a658_2    conda-forge
aws-checksums             0.1.17               h184a658_2    conda-forge
aws-crt-cpp               0.24.2               ha28989d_2    conda-forge
aws-sdk-cpp               1.10.57             hec69fbc_24    conda-forge
blas                      1.0                         mkl    defaults
blosc                     1.21.4               h0f2a231_0    conda-forge
brotli                    1.1.0                hb9d3cd8_2    conda-forge
brotli-bin                1.1.0                hb9d3cd8_2    conda-forge
brotli-python             1.0.9            py39h6a678d5_8    defaults
bzip2                     1.0.8                h5eee18b_6    defaults
c-ares                    1.32.3               h4bc722e_0    conda-forge
ca-certificates           2024.9.24            h06a4308_0    defaults
certifi                   2024.8.30        py39h06a4308_0    defaults
charset-normalizer        3.3.2              pyhd3eb1b0_0    defaults
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.3.0            py39h74842e3_2    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
deeph                     0.2.2                    pypi_0    pypi
deeph-e3                  0.1                      pypi_0    pypi
dill                      0.3.9                    pypi_0    pypi
e3nn                      0.3.5                    pypi_0    pypi
fonttools                 4.54.1           py39h8cd3c5a_0    conda-forge
freetype                  2.10.4               h0708190_1    conda-forge
frozenlist                1.5.0                    pypi_0    pypi
fsspec                    2024.10.0                pypi_0    pypi
future                    0.18.3           py39h06a4308_0    defaults
giflib                    5.2.2                hd590300_0    conda-forge
gmp                       6.3.0                hac33072_2    conda-forge
gmpy2                     2.1.5            py39h7196dd7_2    conda-forge
googledrivedownloader     0.4                      pypi_0    pypi
grpcio                    1.66.2                   pypi_0    pypi
h5py                      3.12.1                   pypi_0    pypi
hdf4                      4.2.15               h9772cbc_5    conda-forge
hdf5                      1.14.3          nompi_hdf9ad27_105    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.7              py39h06a4308_0    defaults
importlib-resources       6.4.5              pyhd8ed1ab_0    conda-forge
importlib_resources       6.4.5              pyhd8ed1ab_0    conda-forge
intel-openmp              2021.4.0          h06a4308_3561    defaults
isodate                   0.6.1                    pypi_0    pypi
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h0b41bf4_3    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
kiwisolver                1.4.7            py39h74842e3_0    conda-forge
krb5                      1.20.1               h81ceb04_0    conda-forge
latexcodec                2.0.1              pyh9f0ad1d_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.40                 h12ee557_0    defaults
lerc                      3.0                  h9c3ff4c_0    conda-forge
libaec                    1.1.3                h59595ed_0    conda-forge
libblas                   3.9.0            12_linux64_mkl    conda-forge
libbrotlicommon           1.1.0                hb9d3cd8_2    conda-forge
libbrotlidec              1.1.0                hb9d3cd8_2    conda-forge
libbrotlienc              1.1.0                hb9d3cd8_2    conda-forge
libcblas                  3.9.0            12_linux64_mkl    conda-forge
libcurl                   8.9.1                h251f7ec_0    defaults
libdeflate                1.10                 h7f98852_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libffi                    3.4.4                h6a678d5_1    defaults
libgcc                    14.2.0               h77fa898_1    conda-forge
libgcc-ng                 14.2.0               h69a702a_1    conda-forge
libgfortran               14.1.0               h69a702a_1    conda-forge
libgfortran-ng            14.1.0               h69a702a_1    conda-forge
libgfortran5              14.1.0               hc5f4f2c_1    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
liblapack                 3.9.0            12_linux64_mkl    conda-forge
libnetcdf                 4.9.2           nompi_h9db66fd_110    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.0                h5eee18b_0    defaults
libopenblas               0.3.21               h043d6bf_0    defaults
libpng                    1.6.37               h21135ba_2    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libstdcxx-ng              14.2.0               h4852527_1    conda-forge
libtiff                   4.3.0                h0fcbabc_4    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libwebp                   1.2.2                h3452ae3_0    conda-forge
libwebp-base              1.2.2                h7f98852_1    conda-forge
libxc                     6.2.2            cpu_h5ef0af7_6    conda-forge
libxc-c                   6.2.2            cpu_h20a523f_6    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxml2                   2.12.7               hc051c1a_1    conda-forge
libzip                    1.10.1               h2629f0a_3    conda-forge
libzlib                   1.2.13               h4ab18f5_6    conda-forge
lua                       5.4.6                h9627803_0    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
markdown                  3.7                      pypi_0    pypi
markupsafe                3.0.2                    pypi_0    pypi
matplotlib-base           3.9.2            py39hbfdbfaf_0    defaults
mkl                       2021.4.0           h06a4308_640    defaults
mkl-service               2.4.0            py39h7f8727e_0    defaults
mkl_fft                   1.3.1            py39hd3c417c_0    defaults
mkl_random                1.2.2            py39h51133e4_0    defaults
monty                     2024.7.30          pyhd8ed1ab_0    conda-forge
mpc                       1.3.1                h24ddda3_1    conda-forge
mpfr                      4.2.1                h90cbb55_3    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
multidict                 6.1.0                    pypi_0    pypi
multiprocess              0.70.17                  pypi_0    pypi
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
ncurses                   6.4                  h6a678d5_0    defaults
netcdf-fortran            4.6.1           nompi_h22f9119_106    conda-forge
networkx                  3.2.1              pyhd8ed1ab_0    conda-forge
numpy                     1.24.3           py39h14f4228_0    defaults
numpy-base                1.24.3           py39h31eccc5_0    defaults
openjpeg                  2.5.0                h7d73246_0    conda-forge
openssl                   3.3.2                hb9d3cd8_0    conda-forge
opt-einsum                3.4.0                    pypi_0    pypi
opt-einsum-fx             0.1.4                    pypi_0    pypi
packaging                 24.1             py39h06a4308_0    defaults
palettable                3.3.3              pyhd8ed1ab_0    conda-forge
pandas                    2.2.3            py39h3b40f6f_1    conda-forge
pathos                    0.3.3                    pypi_0    pypi
pillow                    9.1.1            py39hae2aec6_0    conda-forge
pip                       24.2             py39h06a4308_0    defaults
platformdirs              3.10.0           py39h06a4308_0    defaults
plotly                    5.24.1             pyhd8ed1ab_0    conda-forge
pooch                     1.7.0            py39h06a4308_0    defaults
pox                       0.3.5                    pypi_0    pypi
ppft                      1.7.6.9                  pypi_0    pypi
propcache                 0.2.0                    pypi_0    pypi
protobuf                  5.28.2                   pypi_0    pypi
psutil                    6.0.0                    pypi_0    pypi
pthread-stubs             0.4               hb9d3cd8_1002    conda-forge
pybind11-abi              4                    hd3eb1b0_1    defaults
pybtex                    0.24.0             pyhd8ed1ab_2    conda-forge
pyg-lib                   0.4.0+pt112cu116          pypi_0    pypi
pymatgen                  2024.8.9         py39hb9c10d8_0    conda-forge
pyparsing                 3.1.4              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py39h06a4308_0    defaults
python                    3.9.12          h2660328_1_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-louvain            0.16                     pypi_0    pypi
python-tzdata             2024.2             pyhd8ed1ab_0    conda-forge
python_abi                3.9                      5_cp39    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0.2            py39h8cd3c5a_1    conda-forge
rdflib                    7.0.0                    pypi_0    pypi
readline                  8.2                  h5eee18b_0    defaults
requests                  2.32.3           py39h06a4308_0    defaults
ruamel.yaml               0.18.6           py39hd1e30aa_0    conda-forge
ruamel.yaml.clib          0.2.8            py39hd1e30aa_0    conda-forge
s2n                       1.3.54               h06160fa_0    conda-forge
scikit-learn              1.5.2                    pypi_0    pypi
scipy                     1.13.1                   pypi_0    pypi
setuptools                75.1.0           py39h06a4308_0    defaults
siesta                    5.0.1           nompi_ha48f1ba_1001    conda-forge
six                       1.16.0             pyhd3eb1b0_1    defaults
snappy                    1.1.10               hdb0a2a9_1    conda-forge
spglib                    2.5.0            py39h3ccf219_2    conda-forge
sqlite                    3.38.2               hc218d9a_0    defaults
sympy                     1.13.3          pypyh2585a3b_103    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tenacity                  9.0.0              pyhd8ed1ab_0    conda-forge
tensorboard               2.18.0                   pypi_0    pypi
tensorboard-data-server   0.7.2                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tk                        8.6.12               h27826a3_0    conda-forge
torch                     1.12.0+cu116             pypi_0    pypi
torch-cluster             1.6.0+pt112cu116          pypi_0    pypi
torch-geometric           2.6.1                    pypi_0    pypi
torch-scatter             2.1.0+pt112cu116          pypi_0    pypi
torch-sparse              0.6.16+pt112cu116          pypi_0    pypi
torch-spline-conv         1.2.1+pt112cu116          pypi_0    pypi
torchvision               0.13.0+cu116             pypi_0    pypi
tqdm                      4.66.5             pyhd8ed1ab_0    conda-forge
typing-extensions         4.11.0           py39h06a4308_0    defaults
typing_extensions         4.11.0           py39h06a4308_0    defaults
tzdata                    2024b                h04d1e81_0    defaults
uncertainties             3.2.2              pyhd8ed1ab_1    conda-forge
unicodedata2              15.1.0           py39hd1e30aa_0    conda-forge
urllib3                   2.2.3            py39h06a4308_0    defaults
werkzeug                  3.0.4                    pypi_0    pypi
wheel                     0.44.0           py39h06a4308_0    defaults
xdeeph                    0.1                      pypi_0    pypi
xorg-libxau               1.0.11               hb9d3cd8_1    conda-forge
xorg-libxdmcp             1.1.5                hb9d3cd8_0    conda-forge
xz                        5.4.6                h5eee18b_1    defaults
yaml                      0.2.5                h7f98852_2    conda-forge
yarl                      1.17.1                   pypi_0    pypi
zipp                      3.20.2             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h4ab18f5_6    conda-forge
zstd                      1.5.2                h8a70e8d_1    conda-forge

**Comment by chtchelkatchev** [comment]
I have the same problem with centos os 7.

**Comment by rossfu7** [comment]
I have the same problem with my AWS EC2 instance running Amazon Linux 2. 
Note: If I type `\lib64\libc.so.6` on my console I see the text "GNU C Library (GNU libc) stable release version 2.26, by Roland McGrath et al.", which means that I have GLIBC_2.26 installed (I think). It would be good to know if I can downgrade pyg to a compatible version, similar to issue #367.

**Comment by So-cean** [comment]
same problem

**Comment by akihironitta** [comment]
Hi everyone, sorry about the delay. 

We're actively working on supporting glibc 2.28+, but we have no plan to spport glibc 2.27 as the latest stable PyTorch 2.7 release doesn't support it either.

For those people seeing the version `GLIBC_2.28` or later in your error message, we're planning to add support for older environments via #457 hopefully soon.
